packages/base/room.gts

```diff
@@ -246,6 +247,14 @@ class PatchField extends FieldDef {
   @field payload = contains(PatchObjectField);
 }
 
+// A map from card document hash to the first card fragment event id.
+// This map can be used to avoid sending the same version of the card more than once in a conversation.
+// We can reuse exisiting eventId if user attached the same version of the card.
+const cardDocToEventId: Map<string, string> = new Map();
+export function retrieveEventId(cardDoc: LooseSingleCardDocument) {
+  return cardDocToEventId.get(ethers.keccak256(JSON.stringify(cardDoc)));
```

Comment:
let's not use this hash, its way overkill and introduces all sorts of dependencies that we don't need in our app. rather, let's just use an md5 checksum. it's much faster and doesn't require all the dependencies of `ethers`